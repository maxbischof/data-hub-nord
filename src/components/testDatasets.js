export const datasets = [
  {
    id: 1,
    name: "Arbeitslosigkeit Kiel",
    description: "Arbeitslosigkeit in Kiel absolut und relativ seit 2009.",
    url:
      "https://www.kiel.de/opendata/kiel_wirtschaft_arbeit_unterbeschaeftigung.csv",
    imageUrl:
      "https://images.unsplash.com/photo-1529400971008-f566de0e6dfc?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=80",
    keys: [],
    license: "Datenlizenz Deutschland – Zero – Version 2.0",
    publisher: "Landeshauptstadt Kiel",
    displayedAs: "",
    seperator: ";"
  },
  {
    id: 2,
    name: "Badegewässer Messungen",
    description: "Messungen der Badegewässerqualität in Schleswig Holstein.",
    url:
      "http://efi2.schleswig-holstein.de/bg/opendata/v_proben_odata.csv",
    imageUrl:
      "https://images.unsplash.com/photo-1527694545342-6176693ca513?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60",
    keys: [
      'BADEGEWAESSERID',
      'MESSSTELLENNAME',
      'MESSSTELLENID',
      'UEBERWACHUNGSARTID',
      'UEBERWASCHUNGSARTTEXT',
      'GEWAESSERKATEGORIE',
      'KUESTENGEWAESSER',
      'PROBEID',
      'DATUMMESSUNG',
      'PROBENART',
      'ECOLI',
      'INTEST_ENTEROKOKKEN',
      'WASSERTEMP',
      'LUFTTEMP',
      'SICHTTIEFE',
      'BEMERKUNG'],
    license: "Datenlizenz Deutschland – Zero – Version 2.0",
    publisher: "Landeshauptstadt Kiel",
    displayedAs: "",
    seperator: "|"
  },
]
