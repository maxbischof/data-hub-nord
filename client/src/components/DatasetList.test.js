import React, { useRef } from 'react'
import renderer from 'react-test-renderer'
import DataList from './DatasetList'
import { datasets } from './testDatasets'
import { BrowserRouter as Router } from 'react-router-dom'

it('renders correctly', () => {
  const tree = renderer.create(
    <Router>
      <DataList
        datasets={datasets}
        headline={'Alle DatensÃ¤tze'}
        showMoreButton={true}
      />
    </Router>
  )
  expect(tree).toMatchSnapshot()
})
