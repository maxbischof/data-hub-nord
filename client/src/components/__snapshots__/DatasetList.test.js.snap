// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h2
    className="sc-iCoHVE pBahF"
  />,
  <section
    className="sc-gKAblj dzbQLo"
  >
    <a
      href="/datensaetze/1"
      onClick={[Function]}
    >
      <article
        className="sc-bdnylx ccOnhP"
      >
        <h3
          className="sc-gtssRu cyIxrX"
        >
          Arbeitslosigkeit Kiel
        </h3>
        <hr />
        <p
          className="sc-dlnjPT hCTrCU"
        >
          Arbeitslosigkeit in Kiel absolut und relativ seit 2009.
        </p>
      </article>
    </a>
    <a
      href="/datensaetze/2"
      onClick={[Function]}
    >
      <article
        className="sc-bdnylx ccOnhP"
      >
        <h3
          className="sc-gtssRu cyIxrX"
        >
          Badegewässer Messungen
        </h3>
        <hr />
        <p
          className="sc-dlnjPT hCTrCU"
        >
          Messungen der Badegewässerqualität in Schleswig Holstein.
        </p>
      </article>
    </a>
  </section>,
]
`;
